src/textual/widget.py

```diff
@@ -460,6 +500,20 @@ def mount(
             provided a ``MountError`` will be raised.
         """
 
+        # Check for duplicate IDs in the incoming widgets
+        ids_to_mount = [widget.id for widget in widgets if widget.id is not None]
+        unique_ids = set(ids_to_mount)
+        num_unique_ids = len(unique_ids)
+        num_widgets_with_ids = len(ids_to_mount)
+        if num_unique_ids != num_widgets_with_ids:
```

Comment:
Bravo for detailed error messages. I have a feeling this may be overkill though. Would it be enough to say "Can't mount widgets with duplicate IDs" ?

Reply:
I withdraw my objection.